File: go/0102M-二叉树的层序遍历.go
/*
 * @Author: SukiEva
 * @Date: 2021-11-02 21:41:59
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */

package main

import "container/list"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func levelOrder(root *TreeNode) [][]int {
	ans := [][]int{}
	if root == nil {
		return ans
	}
	queue := list.New()
	queue.PushBack(root)
	t := make([]int, 0)
	for queue.Len() > 0 {
		l := queue.Len()
		for i := 0; i < l; i++ {
			node := queue.Remove(queue.Front()).(*TreeNode)
			t = append(t, node.Val)
			if node.Left != nil {
				queue.PushBack(node.Left)
			}
			if node.Right != nil {
				queue.PushBack(node.Right)
			}
		}
		ans = append(ans, t)
		t = []int{}
	}
	return ans
}

File: cpp/0102M-二叉树的层序遍历.cpp
/*
 * @Author: SukiEva
 * @Date: 2021-12-21 09:19:13
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
#include <bits/stdc++.h>

#include "include/ListNode.h"
#include "include/TreeNode.h"

using namespace std;

class Solution {
   public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ans;
        if (root == nullptr) return ans;
        queue<TreeNode*> que;
        vector<int> tmp;
        que.push(root);
        while (!que.empty()) {
            int l = que.size();
            while (l--) {
                TreeNode* node = que.front();
                que.pop();
                tmp.push_back(node->val);
                if (node->left != nullptr) que.push(node->left);
                if (node->right != nullptr) que.push(node->right);
            }
            ans.push_back(tmp);
            tmp.clear();
        }
        return ans;
    }
};